sor/src/main/java/com/bazaarvoice/emodb/sor/db/astyanax/CqlDataWriterDAO.java

```diff
@@ -0,0 +1,318 @@
+        ConsistencyLevel consistencyLevel = SorConsistencies.toCql(consistency);
+
+        ResultSetFuture oldTableFuture = null;
+
+        // this write statement should be removed in the next version
+        if(_writeToOld) {
+            TableDDL deltaTableDDL = placement.getDeltaTableDDL();
+            Statement oldTableStatement = QueryBuilder.insertInto(deltaTableDDL.getTableMetadata())
+                    .value(deltaTableDDL.getRowKeyColumnName(), rowKey)
+                    .value(deltaTableDDL.getChangeIdColumnName(), compactionKey)
+                    .value(deltaTableDDL.getValueColumnName(), encodedCompaction)
+                    .setConsistencyLevel(consistencyLevel);
+            oldTableFuture =  session.executeAsync(oldTableStatement);
+        }
+
+
+        if (_writeToNew) {
+            // create new atomic batch
+            BlockedDeltaTableDDL blockedDeltaTableDDL = placement.getBlockedDeltaTableDDL();
+            BatchStatement newTableStatement = new BatchStatement(BatchStatement.Type.LOGGED);
+            insertBlockedDeltas(newTableStatement, blockedDeltaTableDDL, consistencyLevel, rowKey, compactionKey, encodedBlockedCompaction);
+            session.executeAsync(newTableStatement).getUninterruptibly();
+        }
+
+        // Wait for both statements to return
+        if (oldTableFuture != null) {
+            oldTableFuture.getUninterruptibly();
+        }
+
+    }
+
+    private Statement deleteStatement(TableDDL tableDDL, ByteBuffer rowKey, UUID changeId, ConsistencyLevel consistencyLevel) {
+        return QueryBuilder.delete()
+                .from(tableDDL.getTableMetadata())
+                .where(eq(tableDDL.getRowKeyColumnName(), rowKey))
+                .and(eq(tableDDL.getChangeIdColumnName(), changeId))
+                .setConsistencyLevel(consistencyLevel);
+    }
+
+    private void deleteCompactedDeltas(ByteBuffer rowKey, WriteConsistency consistency, DeltaPlacement placement,
+                                       CassandraKeyspace keyspace, Collection<UUID> changesToDelete,
+                                       List<History> historyList) {
+
+        Session session = keyspace.getCqlSession();
+        ConsistencyLevel consistencyLevel = SorConsistencies.toCql(consistency);
+
+        // the old statement should be removed in the next version
+        BatchStatement oldTableStatement = _writeToOld ? new BatchStatement(BatchStatement.Type.UNLOGGED) : null;
+        BatchStatement newTableStatement = _writeToNew ? new BatchStatement(BatchStatement.Type.UNLOGGED) : null;
+        ResultSetFuture oldTableFuture = null;
+
+        if (historyList != null && !historyList.isEmpty()) {
+            AuditBatchPersister auditBatchPersister = CqlAuditBatchPersister.build(_writeToOld ? oldTableStatement : newTableStatement,
+                    placement.getDeltaHistoryTableDDL(), _changeEncoder, _auditStore);
+            _auditStore.putDeltaAudits(rowKey, historyList, auditBatchPersister);
```

Comment:
I made a similar comment in PR #116 : Should the history deltas be attached to either the old or new batch statement?  Since they go to a different Cassandra table should they go in their own separate request?

Reply:
done